dapp/client/App.elm

```diff
@@ -67,19 +73,33 @@ update : Message -> Model -> ( Model, Cmd Message )
 update message model =
     case message of
         AddressChanged newAddress ->
-            ( { model | address = Just newAddress }, Cmd.none )
+            ( { initialModel | address = Just newAddress }, Cmd.none )
+
+        WalletChanged newWallet ->
+            ( { model | wallet = Just newWallet, eth = Nothing }, Cmd.none )
```

Comment:
That's an excellent point. I actually was thinking about that this morning and realized that resetting to initialModel plus address when address changes was much nicer than what I did for WalletChanged.